### ISSUE TITLE
Multithread support?

### ISSUE THREAD

**Issue by ericvon11** [satijalab/seurat#331]
I'm a novice at this, so this might not be possible at all. Though as I understand it, using R to run C++ has the added benefit of multithread support. As much of the 'heavy lifting' in Seurat is done via cpp, I was curious to know if there's any way to add multithread support to speed up functions that currently take a very long time to run. ScaleData can often take a day or two when trying to scale all genes. Rtsne also has openmp support, so any ability to specify threads would help speed this up drastically.

So in short, I was curious to know if its possible to add multithread support to the Seurat code to utilize processors better and speed up some of the computationally more intensive functions.

Thanks,
Eric

**Comment by andrewwbutler** [comment]
Hi Eric,

It is definitely possible to add multithread/multicore support and it's something we'd like to do in the future. 

The ScaleData step (assuming you're regressing out some variable) that you're referring to is slow because it's building models for every gene you specify, which is done in R and can take a bit of time. This is why we often recommend only scaling a subset of genes here but the procedure does potentially lend itself nicely to parallelization. We'd be happy to consider a PR if you'd like to implement a parallelized version of any of these functions.
